battle-engine.js

```diff
@@ -3466,7 +3466,12 @@ class Battle extends Tools.BattleDex {
 
 		this.makeRequest('move');
 
-		if (Tools.getFormat(this.format).isWildEncounter) this.add('raw', '<button name="send" value="/throwpokeball pokeball" class="button">Throw a pokeball</button>');
+		if (Tools.getFormat(this.format).isWildEncounter) {
+			 let balls = ['pokeball', 'greatball', 'ultraball', 'masterball'];
+			 balls.forEach(pokeball => {
+				this.add('raw', '<button name="send" value="/throwpokeball ' + pokeball + '" style="background:transparent;border:none;"><img src="http://www.serebii.net/itemdex/sprites/pgl/' + pokeball + '.png" width="30" height="30"></button>&nbsp;&nbsp;');
+			});
```

Comment:
Ive told you 3 times now, send all the buttons in 1 message like this:
```javascript
let buttons = '';
for (let i = 0; i < balls.length; i++) {
  buttons += '<button name="send" value="/throwpokeball ' + balls[i] + '" style="background:transparent;border:none;"><img src="http://www.serebii.net/itemdex/sprites/pgl/' + balls[i] + '.png" width="30" height="30"></button>&nbsp;&nbsp;';
}
this.add('raw', buttons);
```
Also don't use `forEach` on arrays, it has bad performance compared with other options like `for of` or a normal loop.